crates/oxc_minifier/src/peephole/remove_unused_expression.rs

```diff
@@ -910,12 +959,14 @@ mod test {
         test("var a, b; a >= b", "var a, b;");
 
         test_same("var a, b; a + b");
-        test("var a, b; 'a' + b", "var a, b; '' + b");
+        test("var a, b; 'a' + b", "var a, b;");
         test_same("var a, b; a + '' + b");
         test("var a, b, c; 'a' + (b === c)", "var a, b, c;");
-        test("var a, b; 'a' + +b", "var a, b; '' + +b"); // can be improved to "var a, b; +b"
+        test("var a, b; 'a' + +b", "var a, b; +b");
         test_same("var a, b; a + ('' + b)");
         test("var a, b, c; a + ('' + (b === c))", "var a, b, c; a + ''");
+        test("Math.floor(+d / 1000)", "Math.floor(d / 1000)");
+        test("2 - +this._x.call(null, node.data)", "2 - this._x.call(null, node.data)");
```

Comment:
This is not safe unless it is known that `this._x.call(null, node.data)` does not return a BigInt.
```js
2 - +0n // TypeError: Cannot convert a BigInt value to a number
2 - 0n // no error
```

Reply:
Ah, no, forget about this 😅

Reply:
I thinks it's the same in this case

![CleanShot 2025-04-07 at 14 43 03@2x](https://github.com/user-attachments/assets/20197fad-8762-4872-8055-ef46109db96b)